_docker/lib/export/export_diff.rb

```diff
@@ -0,0 +1,124 @@
+          @log.warn("Unable to locate sitemap.xml for current export at expected location '#{sitemap_xml_path}'. Cannot generate list of URLs to clear from cache.")
+        end
+        current_sitemap
+      end
+
+      def load_previous_sitemap
+        previous_export_directory = Dir.glob("#{ @export_archive_directory }/export-*").sort.select{ |directory| File.exist?("#{ directory }/sitemap.xml") }.last
+
+        if previous_export_directory.nil?
+          @log.warn("Unable to perform export comparison as there appear to be no previous exports within the export archive directory '#{ @export_archive_directory }'.")
+          return nil
+        end
+
+        @log.info("Using sitemap.xml from previous export archived at '#{ previous_export_directory }' to determine any modified content.")
+        load_sitemap_xml("#{ previous_export_directory }/sitemap.xml")
+      end
+
+      def load_sitemap_xml(sitemap_xml_path)
+        return nil unless File.exist?(sitemap_xml_path)
+
+        xml_document = nil
+
+        begin
+          xml_document = Nokogiri::XML(File.open(sitemap_xml_path))
+          if xml_document.css('url').size == 0
+            log.warn("A sitemap.xml exists at '#{ sitemap_xml_path }' but it contains 0 <url> entries and cannot be used to determine content that has changed.")
+            return nil
+          end
+        rescue
+          @log.error("Failed to parse or open sitemap.xml at path '#{ sitemap_xml_path }'.")
+          return nil
+        end
+
+        @log.info("Using current sitemap.xml at location '#{sitemap_xml_path}'.")
+        xml_document
+      end
+
+
+      def build_lastmod_hash(sitemap)
+        url_modification_dates = {}
+        sitemap.css('url').each do |url|
+          location = url.css('loc')
+          unless location.empty?
+            last_modification_time = url.css('lastmod')
+            url_modification_dates[location.first.text] = last_modification_time.empty? ? Time.now.to_s : last_modification_time.first.text
+          end
+        end
+
+        url_modification_dates
+      end
+
+
+      def check_for_modified_content(current_sitemap, previous_sitemap)
+
+        previous_urls = build_lastmod_hash(previous_sitemap)
+        current_urls = build_lastmod_hash(current_sitemap)
+
+        modified_urls = []
+
+        current_urls.keys.each do |url|
+          current_timestamp = current_urls[url]
+          previous_timestamp = previous_urls[url]
+
+          if previous_timestamp.nil?
+            @log.info("URL '#{ url }' did not exist in the previous export and is new content. No cache clear required.")
+            next
+          end
+
+          if Date.parse(current_timestamp) > Date.parse(previous_timestamp)
+            @log.info("Content of URL '#{ url }' has changed. <lastmod> in current sitemap '#{ current_timestamp }', <lastmod> in previous sitemap '#{ previous_timestamp }'.")
+            modified_urls.push(url)
+          end
+
+        end
```

Comment:
Feels like there should be some sort of faster way of doing this using some set theory, but my brain isn't working enough to figure that out right now.